### PR TITLE
Cryptography

### PR DESCRIPTION
**Pull request by amsce-iscteiul**
Encrypting, decrypting, confirming certificate, confirming integrity, diffie-hellman key generation .Still not tested!!!

### CODE REVIEW

File: milestone1/src/Cryptography/Crypto.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
+public class Crypto{
+    private static String PATH_TO_CERTIFICATE = "path";
+    private static String HMAC_ALGORITHM = "HmacSHA256";
+    private static String CIFER_ALGORITHM = "AES/CBC/PKCS5Padding";
```
Comment (ALonelyDementor) [inline milestone1/src/Cryptography/Crypto.java]: CIPHER not CIFER
```suggestion
    private static String CIPHER_ALGORITHM = "AES/CBC/PKCS5Padding";
```

**Review by BodCar — CHANGES_REQUESTED** [review]
Missing import statements.

**Review by ALonelyDementor — CHANGES_REQUESTED** [review]
As @BodCar said, imports are missing.

File: milestone1/src/Grid.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    public void addRandomPosition(String user,boolean byzantine){
+        Random rand = new Random();
+        int posx = rand.nextInt(collumns);
+        int posy = rand.nextInt(rows);
+        if(byzantine){
+            userPositions.remove(user);
+            Point p= userPositions.get(user);
+            int prevx = p.x;
+            int prevy = p.y;
+            grid[prevx][prevy].remove(user);
```
Comment (BodCar) [inline milestone1/src/Grid.java]: I don't really understand this. Why are we removing byzantine users from the grid?

File: milestone1/src/Grid.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+        ArrayList<String> nearUsers = new ArrayList<>();
+        Point p = userPositions.get(user);
+        int posx = p.x;
+        int posy = p.y;
+
+        for (int r = p.x - maxDist; r < rows; r++) {
+            if (r < 0)
+                r = 0;
+            for (int c = p.y - maxDist; c < collumns; c++) {
```
Comment (BodCar) [inline milestone1/src/Grid.java]: Why are we defining posx and posy if we are not using them? We should use them on the for loops.

**Comment by ALonelyDementor** [comment]
@amsce-iscteiul will you still create the signature class?

**Pull request merged**